common/planners/src/plan_node_rewriter.rs

```diff
@@ -432,6 +446,30 @@ pub trait PlanRewriter: Sized {
         Ok(PlanNode::ShowDatabases(plan.clone()))
     }
 
+    fn rewrite_show_engines(&mut self, plan: &ShowEnginesPlan) -> Result<PlanNode> {
+        Ok(PlanNode::ShowEngines(plan.clone()))
+    }
+
+    fn rewrite_show_functions(&mut self, plan: &ShowFunctionsPlan) -> Result<PlanNode> {
```

Comment:
Fatter version may cause the code bloat like these.

Reply:
We don't have many `show` command, and add onme show command is easy, I think it's ok.

For the plan_node_rewriter, [common/planners/src/plan_node_rewriter.rs](https://github.com/datafuselabs/databend/pull/4130/files/2fd62a917f4e726ccf25c1fc200cce59712d2cfd#diff-5c0a6979f46d0a3071e2d231fec9dad84867b12c7e690f972954408d7adf594e) there are many empty rewrite, maybe we should do something in it.